src/axom/quest/tests/quest_intersection_shaper.cpp

```diff
@@ -0,0 +1,553 @@
+}
+#endif
+#if defined(AXOM_USE_HIP)
+TEST(IntersectionShaperTest, case1_hip)
+{
+  constexpr double tolerance = 1.e-10;
+  replacementRuleTestSet(case1, "hip", quest::IntersectionShaper::hip, tolerance);
+}
+#endif
+
+// case2
+#if defined(AXOM_SEQ_INTERSECTION_SHAPER_WORKS)
+TEST(IntersectionShaperTest, case2_seq)
+{
+  constexpr double tolerance = 1.e-10;
+  replacementRuleTestSet(case2, "seq", quest::IntersectionShaper::seq, tolerance);
+}
+#endif
+#if defined(AXOM_USE_OPENMP)
+TEST(IntersectionShaperTest, case2_omp)
+{
+  constexpr double tolerance = 1.e-10;
+  replacementRuleTestSet(case2, "omp", quest::IntersectionShaper::omp, tolerance);
+}
+#endif
+#if defined(AXOM_USE_CUDA)
+TEST(IntersectionShaperTest, case2_cuda)
+{
+  constexpr double tolerance = 1.e-10;
+  replacementRuleTestSet(case2, "cuda", quest::IntersectionShaper::cuda, tolerance);
+}
+#endif
+#if defined(AXOM_USE_HIP)
+TEST(IntersectionShaperTest, case2_hip)
+{
+  constexpr double tolerance = 1.e-10;
+  replacementRuleTestSet(case2, "hip", quest::IntersectionShaper::hip, tolerance);
+}
+#endif
+
+// case3
+#if defined(AXOM_SEQ_INTERSECTION_SHAPER_WORKS)
+TEST(IntersectionShaperTest, case3_seq)
+{
+  constexpr double tolerance = 1.e-10;
+  replacementRuleTestSet(case3, "seq", quest::IntersectionShaper::seq, tolerance);
+}
+#endif
+#if defined(AXOM_USE_OPENMP)
+TEST(IntersectionShaperTest, case3_omp)
+{
+  constexpr double tolerance = 1.e-10;
+  replacementRuleTestSet(case3, "omp", quest::IntersectionShaper::omp, tolerance);
+}
+#endif
+#if defined(AXOM_USE_CUDA)
+TEST(IntersectionShaperTest, case3_cuda)
+{
+  constexpr double tolerance = 1.e-10;
+  replacementRuleTestSet(case3, "cuda", quest::IntersectionShaper::cuda, tolerance);
+}
+#endif
+#if defined(AXOM_USE_HIP)
+TEST(IntersectionShaperTest, case3_hip)
+{
+  constexpr double tolerance = 1.e-10;
+  replacementRuleTestSet(case3, "hip", quest::IntersectionShaper::hip, tolerance);
+}
+#endif
+
+// case4
+#if defined(AXOM_SEQ_INTERSECTION_SHAPER_WORKS)
+TEST(IntersectionShaperTest, case4_seq)
+{
+  constexpr double tolerance = 1.e-10;
+  replacementRuleTestSet(case4, "seq", quest::IntersectionShaper::seq, tolerance);
+}
+#endif
+#if defined(AXOM_USE_OPENMP)
+TEST(IntersectionShaperTest, case4_omp)
+{
+  constexpr double tolerance = 1.e-10;
+  replacementRuleTestSet(case4, "omp", quest::IntersectionShaper::omp, tolerance);
+}
+#endif
+#if defined(AXOM_USE_CUDA)
+TEST(IntersectionShaperTest, case4_cuda)
+{
+  constexpr double tolerance = 1.e-10;
+  replacementRuleTestSet(case4, "cuda", quest::IntersectionShaper::cuda, tolerance);
+}
+#endif
+#if defined(AXOM_USE_HIP)
+TEST(IntersectionShaperTest, case4_hip)
+{
+  constexpr double tolerance = 1.e-10;
+  replacementRuleTestSet(case4, "hip", quest::IntersectionShaper::hip, tolerance);
+}
+#endif
+
+int main(int argc, char *argv[])
+{
+  int result = 0;
+#ifdef AXOM_USE_MPI
+  // This is needed because of Axom's c2c reader.
```

Comment:
I recall getting a crash in the PC2CReader which gets instantiated when we ask the shape to load its contour. Since my Axom was built with MPI, the PC2CReader was selected. The first thing it does is call MPI_Comm_rank() and that was crashing. If I remember correctly, it was an MPI_COMM_NULL communicator. Then I made some other changes in the file to set the data collection's comm to MPI_COMM_WORLD. If I recall correctly, the shaper gets its comm from that and PC2CReader gets its comm from the shaper.

Reply:
Thanks -- that makes sense.

I suppose we could check if mpi is initialized here: 
https://github.com/LLNL/axom/blob/1f86e245684480c1d9bcfe88f50011b0bb280e6d/src/axom/quest/interface/internal/QuestHelpers.cpp#L376-L382
and if not, just use the (non-MPI) C2CReader ?
Might be better than requiring MPI for single rank runs.
On the other hand, the current approach works too.